services/util/parseWikiUrl.js

```diff
@@ -21,16 +23,34 @@ function extractName(url) {
   return checks.find(name => !!name);
 }
 
-function splitUrl(url) {
-  if (!regExp.test(url)) {
-    return null;
-  }
-  const matches = url.match(regExp);
+function splitUploadUrl(url) {
+  const matches = url.match(uploadRegExp);
+  const domain = matches[1] === 'commons' ? 'wikimedia' : 'wikipedia';
+  const wikiUrl = `https://${matches[1]}.${domain}.org/`;
+  const segments = url.split('/');
```

Comment:
It seems like it might be a more robust approach to use the [`URL` API](https://nodejs.org/dist/latest-v11.x/docs/api/url.html) to create a (parsed) URL object and then operate on `url.hostname`, `url.pathname`, …

Reply:
Another option would be to copy the code from Lizenzhinweisgenerator as close as possible to the original to make sure we implement the same cases -- From a project-management perspective I would copy the code first, and refactor later (when we have more time and/or money) to reduce the risk of us missing a few url-parsing cases.

Reply:
Side note: If you're using Node.js 10.3+, you can also use named capture groups in regexes:
https://node.green/#ES2018-features--RegExp-named-capture-groups

Reply:
All this is more or less a straight rip off from the existing app - I did not put too much thought into it. So this already more or less follows @tessi's approach 😉